Build fails with Qt 6.9

The KDE CI now has an additional Qt 'next' image currently using 6.9.

qxmpp fails to build with the next version:

```
[   68s] [ 12%] Building CXX object src/CMakeFiles/QXmppQt6.dir/base/QXmppStun.cpp.o
[   68s] cd /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/build/src && /usr/bin/c++ -DBUILD_OMEMO -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050F00 -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_FOREACH -DQT_NO_KEYWORDS -DQT_NO_URL_CAST_FROM_STRING -DQT_USE_QSTRINGBUILDER -DQT_XML_LIB -DQXmppQt6_EXPORTS -DWITH_QCA -I/home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/build/src/QXmppQt6_autogen/include -I/home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base -I/home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/client -I/home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/server -I/home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/build/src -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/gstreamer-1.0 -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/Qca-qt6/QtCrypto -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -O2 -g -DNDEBUG -std=c++20 -fPIC -MD -MT src/CMakeFiles/QXmppQt6.dir/base/QXmppStun.cpp.o -MF CMakeFiles/QXmppQt6.dir/base/QXmppStun.cpp.o.d -o CMakeFiles/QXmppQt6.dir/base/QXmppStun.cpp.o -c /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStun.cpp
[   68s] In file included from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStun_p.h:8,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStun.cpp:7:
[   68s] /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStun.h:128:19: error: field ‘m_attributes’ has incomplete type ‘QSet<short unsigned int>’
[   68s]   128 |     QSet<quint16> m_attributes;
[   68s]       |                   ^~~~~~~~~~~~
[   68s] In file included from /usr/include/qt6/QtCore/qtypeinfo.h:9,
[   68s]                  from /usr/include/qt6/QtCore/qglobal.h:45,
[   68s]                  from /usr/include/qt6/QtCore/qnamespace.h:12,
[   68s]                  from /usr/include/qt6/QtCore/qbytearray.h:9,
[   68s]                  from /usr/include/qt6/QtCore/QByteArray:1,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStanza.h:14,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppIq.h:8,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppJingleData.h:10,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppJingleIq.h:5,
[   68s]                  from /home/abuild/rpmbuild/BUILD/libqxmpp-qt6-1.10.0-build/qxmpp-1.10.0/src/base/QXmppStun.h:8:
[   68s] /usr/include/qt6/QtCore/qcontainerfwd.h:31:29: note: declaration of ‘class QSet<short unsigned int>’
[   68s]    31 | template <typename T> class QSet;
[   68s]       |                             ^~~~
[   68s] make[2]: *** [src/CMakeFiles/QXmppQt6.dir/build.make:970: src/CMakeFiles/QXmppQt6.dir/base/QXmppStun.cpp.o] Error 1
```



(https://build.opensuse.org/package/live_build_log/KDE:Qt:6.9/libqxmpp:qt6/openSUSE_Tumbleweed/x86_64)

--- a/src/base/QXmppStun.h
+++ b/src/base/QXmppStun.h
@@ -9,6 +9,7 @@
 #include "QXmppLogger.h"
 
 #include <QObject>
+#include <QSet>
 
 class CandidatePair;
 class QDataStream;

